### PR TITLE
Add validation when generating data

### PR DESCRIPTION
**Pull request by ludeeus**
This adds usage of the validation schemas for generating the new data.

### CODE REVIEW

File: scripts/data/common.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""Common helpers for data."""
+import sys
+from typing import Any
+import voluptuous as vol
+
+def expand_and_humanize_error(content: dict[str, Any], error: vol.Error) -> list[str] | str:
```
Comment (emontnemery) [inline scripts/data/common.py:6]: `error` is too broadly typed; `vol.Error` is just an `Exception`.
I think we should narrow it to `vol.Invalid`, which has `path` etc., both here and in the scripts which catches voluptuous errors and then calls this function.

File: scripts/data/common.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""Common helpers for data."""
+from __future__ import annotations
+
+import sys
+from typing import Any
+
+import voluptuous as vol
+
+
+def expand_and_humanize_error(content: dict[str, Any], error: vol.Invalid|vol.MultipleInvalid) -> list[str] | str:
```
Comment (emontnemery) [inline scripts/data/common.py:10]: `vol.MultipleInvalid` is a subclass of `vol.Invalid`:
```suggestion
def expand_and_humanize_error(content: dict[str, Any], error: vol.Invalid) -> list[str] | str:
```

Comment (ludeeus) [reply scripts/data/common.py]: Right 👍

File: scripts/data/common.py [hunk 3]
```diff
@@ -1,13 +1,11 @@
 """Common helpers for data."""
-from __future__ import annotations
-
```
Comment (emontnemery) [inline scripts/data/common.py]: I'd still keep this even though it may not be strictly needed.

**Review by emontnemery — APPROVED** [review]
Nice!

**Pull request merged**